### PR TITLE
add official nest compile step & remove now-redundant decorators

### PR DESCRIPTION
**Pull request by chaunceyau**
Love the starter pack - great job! Have you considered cleaning up the models/etc by using the official plugin that auto-adds them? I only provided one example in this PR, but I'm willing to update all. Not sure if there was reason you didn't use! 

Talks about it in NestJS 7 announcement here:
https://trilon.io/blog/announcing-nestjs-7-whats-new#NestJS-GraphQL-Plugin

And also here in the documentation (under CLI plugin🤔): 
https://docs.nestjs.com/graphql/resolvers#cli-plugin

### CODE REVIEW

File: src/models/user.ts [hunk 1]
```diff
@@ -14,20 +14,10 @@ registerEnumType(Role, {
...
-  @Field((type) => String, { nullable: true })
   firstname?: string;
-
-  @Field((type) => String, { nullable: true })
   lastname?: string;
-
-  @Field((type) => Role)
   role: Role;
-
-  @Field((type) => [Post])
```
Comment (marcjulian) [inline src/models/user.ts:29]: I think you still have to tell the compiler about the `Post` type

Comment (chaunceyau) [reply src/models/user.ts:29]: I honestly don't believe so - but definitely correct me if I'm wrong. The examples linked above (from nest 7 announcement/docs) have the following example:

image

image

Comment (marcjulian) [reply src/models/user.ts:29]: Okay cool, if the schema is correctly generated then its perfect.

**Comment by edbond88** [comment]
I checked it works fine!

For the plugin to work, you need to rename files to a specific format: 
`['.input.ts', '.args.ts', '.entity.ts', '.model.ts']`

Or specify your own in the options. But I would prefer standard masks.

**Comment by chaunceyau** [comment]
Yes ^ this is true.

image

File: src/models/user.ts [hunk 2]
```diff
@@ -14,20 +14,10 @@ registerEnumType(Role, {
...
   posts: Post[];
-
   password: string;
```
Comment (marcjulian) [inline src/models/user.ts:22]: Password also needs to be hidden from the GraphQL schema.

Comment (fabiofdsantos) [reply src/models/user.ts:22]: Yeah, just need to add `@HideField()` from `@nestjs/graphql`.

**Comment by edbond88** [comment]
I tried using the plugin and refused.
Because of him, I very often had errors that were not in the files. There was even an error in a file which no longer existed.

**Comment by marcjulian** [comment]
@chaunceyau are migrating all remaining models in this PR? Then I can merge it into master :)

**Comment by fabiofdsantos** [comment]
@edbond88 what do you mean by errors that were not in the files? I've used the plugin successfully after introduced some additional changes to the project.

**Comment by chaunceyau** [comment]
@marcjulian Sorry - I completely put that off. I'll handle it this afternoon/tomorrow morning 👍

**Comment by marcjulian** [comment]
No worries!

**Comment by marcjulian** [comment]
@chaunceyau thanks for your work! 👍 I finished the remaining tasks for the graphql plugin and its available in master now!

**Comment by chaunceyau** [comment]
@marcjulian I really appreciate it - I apologize once again that I never got around to it!

**Comment by marcjulian** [comment]
@chaunceyau no problem :) Please continue to add PRs if you find a missing feature or you can improve something!

**Pull request merged**